nms/app/packages/magmalte/app/views/tracing/TracingDashboard.js

```diff
@@ -0,0 +1,168 @@
+
+import ActionTable from '../../components/ActionTable';
+import CardTitleRow from '../../components/layout/CardTitleRow';
+import CreateTraceButton from './TraceStartDialog';
+import Grid from '@material-ui/core/Grid';
+import HistoryIcon from '@material-ui/icons/History';
+import React from 'react';
+import TopBar from '../../components/TopBar';
+import TraceContext from '../../components/context/TraceContext';
+import WatchIcon from '@material-ui/icons/Watch';
+import withAlert from '@fbcnms/ui/components/Alert/withAlert';
+import {makeStyles} from '@material-ui/styles';
+import {useContext, useState} from 'react';
+import {useEnqueueSnackbar} from '@fbcnms/ui/hooks/useSnackbar';
+
+const useStyles = makeStyles(theme => ({
+  dashboardRoot: {
+    margin: theme.spacing(5),
+  },
+}));
+
+function TracingDashboard() {
+  return (
+    <>
+      <TopBar
+        header="Call Tracing"
+        tabs={[
+          {
+            label: 'Call Traces',
+            to: '/overview',
+            icon: WatchIcon,
+            filters: <div />,
+          },
+        ]}
+      />
+      <TracingTable />
+    </>
+  );
+}
+
+type TracingRowType = {
+  traceID: string,
+  state: 'COMPLETED' | 'ACTIVE',
+  gatewayID: string,
+  traceType: string,
+};
+
+function TracingTableRaw(_: WithAlert) {
+  const [currRow, setCurrRow] = useState<TracingRowType>({});
+  const classes = useStyles();
+  const ctx = useContext(TraceContext);
+  const enqueueSnackbar = useEnqueueSnackbar();
+  const traceMap = ctx.state;
+  const tableData = tracesToRows(traceMap);
+  const tableColumns = [
+    {title: 'Trace ID', field: 'traceID'},
+    {title: 'State', field: 'state'},
+    {title: 'Gateway ID', field: 'gatewayID'},
+    {title: 'Trace Type', field: 'traceType'},
+  ];
+
+  const TraceFilter = () => {
+    return <CreateTraceButton />;
+  };
+
+  const handleStop = async () => {
+    if (currRow.state === 'COMPLETED') {
+      enqueueSnackbar('Call trace ' + currRow.traceID + ' already stopped.', {
+        variant: 'error',
+      });
+      return;
+    }
+
+    try {
+      // $FlowFixMe[prop-missing]: Suppress type error, cannot refine type
+      await ctx.setState?.(currRow.traceID, {
+        requested_end: true,
+      });
+      enqueueSnackbar('Call trace ended successfully', {
+        variant: 'success',
+      });
+    } catch (e) {
+      const errMsg = e.response?.data?.message ?? e.message ?? e;
+      enqueueSnackbar('Failed stopping call trace: ' + errMsg, {
+        variant: 'error',
+      });
+    }
+  };
+
+  return (
+    <div className={classes.dashboardRoot}>
+      <Grid container spacing={4}>
+        <Grid item xs={12}>
+          <CardTitleRow
+            icon={HistoryIcon}
+            label={'Call Traces'}
+            filter={TraceFilter}
+          />
+          <div>
+            <ActionTable
+              data={tableData}
+              columns={tableColumns}
+              handleCurrRow={(row: TracingRowType) => {
```

Comment:
prefer using the shorter notation here - (row : TracingRowType) => setCurrRow(row)